examples/aws/main.tf

```diff
@@ -20,54 +20,69 @@ provider "gardener" {
           namespace: garden-<profile>
         name: garden-<profile>-robot
     current-context: garden-<profile>-robot
-
     EOT*/
 }
 
 resource "gardener_shoot" "test_cluster" {
   metadata {
     name      = "test-cluster"
     namespace = "garden-<profile>"
-
   }
 
   spec {
-    cloud {
-      profile = "aws"
-      region  = "eu-central-1"
+    cloud_profile_name  = "aws"
+    region              = "eu-central-1"
+    secret_binding_name = "<secret_binding>"
 
-      secret_binding_ref {
-        name = "<secret_binding>"
-      }
+    networking {
+      type     = "calico"
+      nodes    = "10.250.0.0/19"
+      pods     = "100.96.0.0/11"
+      services = "100.64.0.0/13"
+    }
 
-      aws {
-        networks {
-          vpc {
-            cidr = "10.250.0.0/16"
+    provider {
+      type = "aws"
+      infrastructure_config {
+        aws {
+          enableecraccess = true
+          networks {
+            vpc {
+              cidr = "10.50.0.0/16"
+            }
+            zones {
+              name     = "eu-central-1a"
+              internal = "10.50.112.0/22"
+              public   = "10.50.96.0/22"
+              workers  = "10.50.0.0/19"
+            }
           }
-
-          internal = ["10.250.112.0/22"]
-          public   = ["10.250.96.0/22"]
-          workers  = ["10.250.0.0/19"]
         }
-
-        worker {
-          name            = "cpu-worker"
-          machine_type    = "m5.large"
-          auto_scaler_min = 3
-          auto_scaler_max = 4
-          max_surge       = 1
-          max_unavailable = 0
-          volume_type     = "gp2"
-          volume_size     = "50Gi"
+      }
+      worker {
+        name            = "cpu-worker"
+        max_surge       = 1
+        max_unavailable = 0
+        maximum         = 2
+        minimum         = 2
+        volume {
+          type = "gp2"
+          size = "30Gi"
+        }
+        machine {
+          type = "t3.medium"
+          image {
+            name    = "coreos"
+            version = "ami-d0dcef3"

```

Comment:
should it not be `ami` as the key as you are passing the ami version and not `version` of the image.